src/components/terms-and-conditions.tsx

```diff
@@ -0,0 +1,205 @@
+// Terms and Conditions Pop-up
+
+import React, { ReactNode, useState } from "react";
+import { Inter } from "next/font/google";
+import { Controller, SubmitHandler, useForm } from "react-hook-form";
+
+const inter = Inter({ subsets: ["latin"] });
+
+type FormValues = {
+  tncAccepted: false;
+};
+
+const Terms_And_Conditions = () => {
+  const {
+    control,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<FormValues>({
+    defaultValues: {
+      tncAccepted: false
+    }
+  });
+
+  const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+    const response = await fetch(`/api/repair-request`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(data)
+    });
+  };
+
+  function PopUp({
+    trigger,
+    children
+  }: {
+    trigger: boolean;
+    children: ReactNode;
+  }) {
+    return trigger ? (
+      <div className="popup justify-item-center fixed inset-6 z-50 flex h-auto overflow-auto rounded-md border border-solid border-black">
+        <div className="popup-inner overflow-auto bg-white bg-opacity-100">
+          <button
+            className="close-btnh mb-4 w-20 border-spacing-0.5 rounded-md border border-solid bg-red-400 text-center text-lg text-white hover:bg-red-300"
+            onClick={() => setShowPopup(false)}
+          >
+            Close
+          </button>
+          {children}
+        </div>
+      </div>
+    ) : null;
+  }
+
+  const [showPopup, setShowPopup] = useState(false);
+  const handleshowPopupChange = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+    setShowPopup((prevState) => !prevState);
+  };
+
+  return (
+    <main
+      className={`flex min-h-screen flex-col items-center justify-between p-24 ${inter.className}`}
+    >
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <label htmlFor="tncAccepted">
+          <div className="relative h-6 justify-center">
+            <Controller
+              control={control}
+              name="tncAccepted"
+              rules={{ required: "*Please refer to the terms and conditions." }}
+              render={({ field: { value, onChange } }) => (
+                <input
+                  className={`${
+                    errors.tncAccepted &&
+                    "border-red-500 focus:border-red-500 focus:ring-red-500"
+                  }`}
+                  type="checkbox"
+                  checked={value}
+                  onChange={(e) => {
+                    onChange(e.target.checked);
+                  }}
+                />
+              )}
+            />
+            <span className="pl-3">
+              I accept the{" "}
+              <button
+                className="text-darkAqua-400"
+                onClick={handleshowPopupChange}
+              >
+                {" "}
+                Terms and Conditions{" "}
+              </button>
+              .
+            </span>
```

Comment:
Is there a reason the button should be in between the text? Perhaps separating the button from the span would be better